src/overlays/actors/ovl_Obj_Blockstop/z_obj_blockstop.c

```diff
@@ -28,12 +29,46 @@ const ActorInit Obj_Blockstop_InitVars = {
     (ActorFunc)NULL,
 };
 
-#endif
+void ObjBlockstop_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjBlockstop* this = THIS;
+    if (Flags_GetSwitch(globalCtx, (s32)this->actor.params)) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    this->actionFunc = ObjBlockstop_CheckCollision;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/ObjBlockstop_Init.s")
+void ObjBlockstop_CheckCollision(ObjBlockstop* this, GlobalContext* globalCtx) {
+    Actor* tempActor;
+    tempActor = globalCtx->actorCtx.actorLists[6].first;
+    while (tempActor) {
+        if ((tempActor->id == 0x7A) && // check if oshihiki (push block)
+            (fabsf(tempActor->world.pos.x - this->actor.world.pos.x) < 20.0f) &&
+            (fabsf(tempActor->world.pos.z - this->actor.world.pos.z) < 20.0f) &&
+            (fabsf(tempActor->world.pos.y - this->actor.world.pos.y) < 20.0f)) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809466F0.s")
+            s32 params = OBJOSHIHIKI_GET_F(tempActor);
+            if (params < 3) {
+                ActorCutscene_SetIntentToPlay((s16)this->actor.cutscene);
+                this->actionFunc = ObjBlockstop_TryPlayCutscene;
+            }
+        }
+        tempActor = tempActor->next;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Blockstop/func_809467E8.s")
+void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
```

Comment:
```suggestion
void ObjBlockstop_TryPlayCutscene(ObjBlockstop* arg0, GlobalContext* arg1) {
```

Any particular reason this isn't `ObjBlockstop_TryPlayCutscene(ObjBlockstop* this, GlobalContext* globalCtx)`?
With that fixed, you should be able to clean up a lot of the casts in this function as well.